scrunch/tests/test_datasets.py

```diff
@@ -1986,6 +1986,178 @@ def test_recode_multiple_response(self):
             }
         })
 
+    def test_create_categorical_missing_case(self):
+        variables = {
+            'var_a': {
+                'id': '001',
+                'alias': 'var_a',
+                'name': 'Variable A',
+                'type': 'numeric',
+                'is_subvar': False
+            },
+            'var_b': {
+                'id': '002',
+                'alias': 'var_b',
+                'name': 'Variable B',
+                'type': 'categorical',
+                'is_subvar': False
+            }
+        }
+        ds_mock = self._dataset_mock(variables=variables)
+        ds = StreamingDataset(ds_mock)
+        kwargs = {
+            'name': 'my mr',
+            'alias': 'mr',
+            'multiple': True,
+            'missing_case': 'var_b == 0',
+            'categories': [
+                {
+                    'id': 1,
+                    'name': 'Facebook',
+                    'case': 'var_a == 1',
+                },
+                {
+                    'id': 2,
+                    'name': 'Twitter',
+                    'case': 'var_b == 1',
+                }
+            ]
+        }
+        with pytest.raises(ValueError) as err:
+            ds.create_categorical(**kwargs)
+        assert 'Entity test_dataset_name has no (sub)variable' in str(err.value) 
+        ds.resource.variables.create.assert_called_with({  
+            'element':'shoji:entity',
```

Comment:
You could probably skip this boilerplate in your test using the shoji wrapper functions